src/game.js

```diff
@@ -0,0 +1,190 @@
+  }
+
+  repeatBtn.removeEventListener("click", playSoundBinded);
+  removeGameEventListeners();
+
+  await new Promise((r) => setTimeout(r, 1200));
+}
+
+function addListeners(id) {
+  const cards = document.querySelectorAll(".card");
+
+  cards.forEach((card, index) => {
+    if (!card.classList.contains("inactive-card")) {
+      card.addEventListener("click", (e) => handleClick(e));
+      if (index === id) {
+        card.addEventListener("click", handleSuccess);
+      } else {
+        card.addEventListener("click", handleFailure);
+      }
+    }
+  });
+}
+
+async function getAnswer(card) {
+  const receiveRightAnswer = getPromiseFromEvent(card);
+  const clickOnMenu = getPromiseFromEvent(document.querySelector("#menu"));
+  const switchModes = getPromiseFromEvent(
+    document.querySelector("#play-train-btn")
+  );
+  const res = await Promise.race([
+    receiveRightAnswer,
+    clickOnMenu,
+    switchModes,
+  ]);
+
+  if (res.id === "menu" || res.id === "play-train-btn") {
+    throw Error("User left the game");
+  }
+}
+
+function getPromiseFromEvent(item, event = "click") {
+  return new Promise((resolve) => {
+    const listener = () => {
+      item.removeEventListener(event, listener);
+      resolve(item);
+    };
+    item.addEventListener(event, listener);
+  });
+}
+
+function handleClick(e) {
+  if (state.gameOn === false) {
+    return;
+  }
+  gameState.clickedCard = e.target.closest(".card");
+}
+
+function handleSuccess() {
+  successSound();
+  makeCardIncative();
+  addStar(true);
+}
+
+function handleFailure() {
+  failureSound();
+  countMistakes();
+  addStar(false);
+}
+
+function addStar(correct) {
+  const container = document.querySelector("#stars-container");
+  const star = document.createElement("p");
+  star.textContent = correct ? "💎" : "💣";
+  star.classList.add("star");
+  container.append(star);
+}
+
+function makeCardIncative() {
+  gameState.clickedCard.removeEventListener("click", handleSuccess);
+  gameState.clickedCard.classList.add("inactive-card");
+}
+
+function countMistakes() {
+  const card = gameState.category.cards[gameState.wordId];
+  incrementMistakes(card);
+  gameState.mistakes++;
+}
+
+function removeGameEventListeners() {
+  document.querySelectorAll(".card").forEach((card) => {
+    card.removeEventListener("click", handleSuccess);
+    card.removeEventListener("click", handleFailure);
+  });
+}
+
+function startGameState(category) {
+  document.querySelector("main").classList.add("game-on");
+  state.gameOn = true;
+  gameState.category = category;
+  gameState.mistakes = 0;
+}
+
+function stopGameState() {
+  document
+    .querySelectorAll(".card")
+    .forEach((card) => card.classList.remove("inactive-card"));
+  document.querySelector("main").classList.remove("game-on");
+  removeGameEventListeners();
+  document.querySelector("#stars-container").textContent = "";
+  state.gameOn = false;
+}
+
+function successSound() {
+  new Audio("./data/audio/correct.mp3").play();
```

Comment:
`Audio().play()` is the `async` method, so we should use it with the `await` command.